### PR TITLE
Rewrite proxy auth header

### PR DESCRIPTION
**Pull request by Malkiz223**
If the first request failed and we changed the proxy with credentials, the old `request.headers['Proxy-Authorization']` will be used instead of creating a new one. Due to this error, we will not be able to connect to the new proxy. This patch fixes this bug.

### CODE REVIEW

**Comment by G000D1ESS** [comment]
Yeah, I have same problem when I rotate proxy with different credentials. 👍

**Comment by wRAR** [comment]
Previous (stalled) PR: https://github.com/scrapy/scrapy/pull/3805

File: tests/test_downloadermiddleware_httpproxy.py [hunk 1]
```diff
@@ -86,6 +86,18 @@ def test_proxy_auth_empty_passwd(self):
         self.assertEqual(req.meta, {'proxy': 'https://proxy:3128'})
         self.assertEqual(req.headers.get('Proxy-Authorization'), b'Basic dXNlcm5hbWU6')
 
+    def test_rewrite_proxy_auth_header_if_use_credentials(self):
+        os.environ['http_proxy'] = 'https://user:pass@proxy:3128'
+        mw = HttpProxyMiddleware()
+        req = Request(
+            'http://scrapytest.org',
+            headers={'Proxy-Authorization': b'Basic dXNlcjo='},
+            meta={'proxy': 'https://user:pass@proxy:3128'},
```
[the remark this answers is not in the document]
Comment (Malkiz223) [reply tests/test_downloadermiddleware_httpproxy.py:95]: Gallaecio, thank you for such a quick response. I agree with your solution to the problem and created a new pull request copying your code if you don't mind

**Comment by Gallaecio** [comment]
[Scrapy 2.6.2 has now been released](https://docs.scrapy.org/en/2.6/news.html#scrapy-2-6-2-2022-07-25), addressing this issue.

For transparency, this is how that happened:

1. I reviewed this pull request, and [I noticed how it would be a backward-incompatible change](https://github.com/scrapy/scrapy/pull/5477#discussion_r850246867).
2. I realized we may need to treat this as a security issue and, against what my review said, actually implement a fix in this direction despite the backward-incompatibility (security triumphs backward compatibility).
3. The Scrapy core team discussed the issue, and we agreed to treat it as a security issue and to address it as part of 2.6.2.
4. It took a while to implement and review the fix, which delayed 2.6.2.

Due to security concerns, I did not want to disclose the reasons for delaying 2.6.2 or not reviewing #5479.

@Malkiz223 We are very grateful that you prepared this pull request, and I am sorry about the extra work you put into #5479 in answer to my review, and my silence since then. I hope that having the change available now, and knowing the big role you played in that, makes up for it to some extent.

**Comment by Malkiz223** [comment]
Thank you very much for your reply and for your work. I was pleasantly surprised to see that both this problem and this pull request were not forgotten.
I am extremely excited to see these changes in the 2.6.2 release, thanks again!

**Pull request closed without merging**